Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/named_function_params.h

```diff
@@ -28,42 +28,93 @@
 
 #define CGAL_PMP_NP_TEMPLATE_PARAMETERS T, typename Tag, typename Base
 #define CGAL_PMP_NP_CLASS CGAL::pmp_bgl_named_params<T,Tag,Base>
-
+namespace CGAL{
+template <typename T, typename Tag, typename Base = boost::no_property>
+struct pmp_bgl_named_params;
+}
+
+#define Cgal_add_pmp_parameter(X_t, X)        \
+namespace CGAL{                               \
+namespace Polygon_mesh_processing{            \
+namespace parameters{                         \
+template<typename K>                          \
+pmp_bgl_named_params<K, X_t>                  \
+X(const K& k)                                 \
+{                                             \
+  typedef pmp_bgl_named_params<K, X_t> Params;\
+  return Params(k);                           \
+}                                             \
+}                                             \
+}                                             \
+}
+
+#define Cgal_add_pmp_parameter_and_enum(X_t, X) \
+  namespace CGAL{                               \
+  enum X_t { X };                               \
+  namespace Polygon_mesh_processing{            \
+  namespace parameters{                         \
+  template<typename K>                          \
+  pmp_bgl_named_params<K, X_t>                  \
+  X(const K& k)                                 \
+  {                                             \
+    typedef pmp_bgl_named_params<K, X_t> Params;\
+    return Params(k);                           \
+  }                                             \
+  }                                             \
+  }                                             \
+}
+
+#define Cgal_add_pmp_parameter_in_struct(X_t, X)       \
+  template<typename K>                                 \
+  pmp_bgl_named_params<K, X_t, self>                   \
+  X(const K& k) const                                  \
+  {                                                    \
+    typedef pmp_bgl_named_params<K, X_t, self> Params; \
+    return Params(k, *this);                           \
+  }
+
+Cgal_add_pmp_parameter_and_enum(geom_traits_t, geom_traits)
```

Comment:
What do you mean "the solution similar to `interface_macros.h` " ? A system that asks each file to define its own macro ?